pkg/resources/mutation/cluster/cluster.go

```diff
@@ -0,0 +1,133 @@
+	Group:    "management.cattle.io",
+	Version:  "v3",
+	Resource: "clusters",
+}
+
+func NewManagementClusterMutator(cache v3.PodSecurityAdmissionConfigurationTemplateCache) *ManagementClusterMutator {
+	return &ManagementClusterMutator{
+		psact: cache,
+	}
+}
+
+// ManagementClusterMutator implements admission.MutatingAdmissionWebhook.
+type ManagementClusterMutator struct {
+	psact v3.PodSecurityAdmissionConfigurationTemplateCache
+}
+
+// GVR returns the GroupVersionKind for this CRD.
+func (m *ManagementClusterMutator) GVR() schema.GroupVersionResource {
+	return managementGVR
+}
+
+// Operations returns list of operations handled by this mutator.
+func (m *ManagementClusterMutator) Operations() []admissionregistrationv1.OperationType {
+	return []admissionregistrationv1.OperationType{admissionregistrationv1.Create, admissionregistrationv1.Update}
+}
+
+// MutatingWebhook returns the MutatingWebhook used for this CRD.
+func (m *ManagementClusterMutator) MutatingWebhook(clientConfig admissionregistrationv1.WebhookClientConfig) *admissionregistrationv1.MutatingWebhook {
+	mutatingWebhook := admission.NewDefaultMutatingWebhook(m, clientConfig, admissionregistrationv1.ClusterScope)
+	mutatingWebhook.SideEffects = admission.Ptr(admissionregistrationv1.SideEffectClassNoneOnDryRun)
+	return mutatingWebhook
+}
+
+// Admit is the entrypoint for the mutator. Admit will return an error if it is unable to process the request.
+func (m *ManagementClusterMutator) Admit(request *admission.Request) (*admissionv1.AdmissionResponse, error) {
+	if request.DryRun != nil && *request.DryRun {
+		return psa.AdmissionResponseAllowed(), nil
+	}
+	oldCluster, newCluster, err := objectsv3.ClusterOldAndNewFromRequest(&request.AdmissionRequest)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get old and new clusters from request: %w", err)
+	}
+	if newCluster.Name == "local" {
+		return psa.AdmissionResponseAllowed(), nil
+	}
+	if newCluster.Spec.RancherKubernetesEngineConfig == nil {
+		// do nothing and leave it to the validator to reject the request
+		return psa.AdmissionResponseAllowed(), nil
+	}
+	newTemplateName := newCluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName
+	oldTemplateName := oldCluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName
+
+	// If the template is set(or changed), update the cluster with the new template's content
+	if newTemplateName != "" {
+		err := m.setPSAConfig(newCluster)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("failed to set PSAconfig: %w", err)
+		}
+	} else {
+		switch request.Operation {
+		case admissionv1.Create:
+			return psa.AdmissionResponseAllowed(), nil
+		case admissionv1.Update:
+			// It is a valid use case where user switches from using PSACT to putting a PluginConfig for PSA under kube-api.AdmissionConfiguration,
+			// but it is not a valid use case where the PluginConfig for PSA has the same content as the one in the previous-set PSACT,
```

Comment:
Same as in the other comment, I'm not sure why _"it is not a valid use case where the PluginConfig for PSA has the same content as the one in the previous-set PSACT"_. Are not allowing the users to "untie" from PSACT and just specify the same content under `kube-api.AdmissionConfiguration`?

Reply:
Yes, the current design is that users can not specify the same content under `kube-api.AdmissionConfiguration` as the previously-set PSACT. This is because we cannot tell if the content under `kube-api.AdmissionConfiguration` is from the previously-set PSACT or users.  Either keeping it when it is from the PSACT or dropping it when it is from the users is not good. 

To solve this problem, we can take the idea from how we support PSA today: adding a setting for enabling/disabling using PSACT in the cluster. However, this also does not provide a much better user experience: users have to configure one more field.